WooCommerce/Classes/ViewRelated/Products/Edit Product/Shipping Settings/ProductShippingSettingsViewController.swift

```diff
@@ -81,13 +94,30 @@ private extension ProductShippingSettingsViewController {
             tableView.register(row.type.loadNib(), forCellReuseIdentifier: row.reuseIdentifier)
         }
     }
+
+    func retrieveProductShippingClass() {
+        guard let shippingClass = shippingClass else {
+            return
+        }
+
+        let action = ProductShippingClassAction
+            .retrieveProductShippingClass(siteID: Int64(product.siteID),
+                                          remoteID: shippingClass.shippingClassID) { [weak self] (shippingClass, error) in
+            self?.shippingClass = shippingClass
```

Comment:
I see `error` is a possible returned param. Should an error check be added here?

Reply:
yea good catch, I just created a ticket to track all the possible errors that we want to handle for release 1: https://github.com/woocommerce/woocommerce-ios/issues/1652

for this error scenario where the Product has a shipping class but we can't fetch it from the server, how about we show an alert when the user taps to edit the shipping class? otherwise, the preselected shipping class in the following list selector screen wouldn't be correct